### ISSUE TITLE
How Serious is This Project?

### ISSUE THREAD

**Issue by machineghost** [chrisabrams/dragon#21]
As a longtime Backbone-user who is looking for more modern features like ES6 and virtual DOM support,  I *love* the concept of Dragon.  And since I'm starting a new project soon, I'm seriously considering using Dragon for it.

However, I'm a bit concerned about the long-term future of the project.  In other words, is this just a side project that you might lose interest in in six months?  Or is it a key part of a real world application you maintain professionally?  Are you committed to evangelizing and supporting Dragon in the future, or is this more of a "here's something cool, feel free to use it, but you're on your own if you do"?

Basically any information you could provide on Dragon's future would be helpful for someone like myself who is considering adopting it.  Thanks.

**Comment by chrisabrams** [comment]
Hey @machineghost,

Great questions, and I come from a pretty similar background so I get where you're coming from. Back at the end of 2014 it was just a fun side project, but after coming across Google's [Incremental DOM](https://github.com/google/incremental-dom) I realized the reality of what the project could be. Since Incremental DOM I've decided to push Dragon pretty hard on projects that I work on. It's the power of virtual dom using Backbone's style but without the JSX crap.

The project is a mix of [Backbone](http://backbonejs.org/) and [Chaplin](http://chaplinjs.org/). I would like to road map breaking out the Chaplin additions such as Collection Views, Dispatcher, Controllers, etc. into optional modules.

I'd recommend checking out the [0.2](https://github.com/chrisabrams/dragon/tree/0.2) branch. I've been very active in it. I would consider the project state still to be alpha (not everything has the level of tests that it should), but I am using 0.2 in production and have an even larger open source project using Dragon that I'll be publishing later this year.

I'm happy to help support anyone who wants to jump into the project, but I'd also want them to realize that I'm the maintainer, not a large community like React. I didn't build this with the intention of "beating React" but I'm not a React fan. It would be nice to have more community support to push this project along.

I think it would be good to jump on a Hangout and discuss what you are planning on working on and see where I can help to make sure you start off on the right foot.

**Comment by chrisabrams** [comment]
I have been planning on showing the project off at meetup groups after 0.2 is finalized. I haven't been pushing for contributors until it was far enough along to be worth someone's time. The project isn't an idea, it's real.

**Comment by machineghost** [comment]
Thanks for the response.  My reason for asking is that I've inherited a legacy codebase that was written with Knockout (and pure evil).  I'm about to start a new (totally separate) app using a new framework.

Initially the app will just be a glorified login page, but over time it will grow, and whatever framework I use for it (once it has "proven itself") will get used to revamp the original app.  So even though I'm only building a small app initially, my decision is essentially "which framework do I want to test-drive for use in all future development?"

My options are:
1. Use Dragon: it has the Backbone base that I love plus full ES6 support and a virtual DOM ... but I risk having to completely replace it in the future  if the project dies (or isn't popular enough to have a plug-in for feature X I need in the future)
2. Use React:  I've done it before, and while I like it less than Backbone, it's sure to still be there (and be acvtively developed with a healthy community) in the future.  Plus no one ever got fired for <s>buying IBM</s> picking a leading framework.
3. Use Angular: not an option for me (Knockout has scared me away from data-binding frameworks for life)

Still not sure which I'll pick, but this:

>I am using 0.2 in production and have an even larger open source project using Dragon that I'll be publishing later this year.

is encouraging.  However, it'd be a lot easier choice if you had a full webpage, CommonJS/AMD/Node packages, articles on Echo JS, etc. :)
